api/controllers/environment.go

```diff
@@ -39,6 +39,18 @@ func (e *EnvironmentController) Routes() []*fireball.Route {
 				"DELETE": e.DeleteEnvironment,
 			},
 		},
+		{
+			Path: "/environment/:id/link",
```

Comment:
This path seems odd given the two ids exist in the request body. maybe `link` should be its own path, like `POST /link` and `DELETE /link`? I'm not sure.

Reply:
It is definitely odd. Redundant information is being passed. But it is consistent. For example, environment update also has `environment_id` in both the path and the body. I think this is a side effect of doing updates as jobs. We are using the same model to capture the job request as the http request body, so the entity id always exists.

If we had 2 models: one for the http put body and one for the update job, we could avoid this.

Reply:
different models is a good idea. We should do that all at once for each entity update though. Add it to that refactor ticket perhaps?